src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -369,6 +380,242 @@ void CascadiaSettings::_LoadDynamicProfiles()
+            lambda2();
+            cv2.wait(lock2);
+
+            // the StorageFolder class has its own methods for obtaining the files within the folder
+            // however, all those methods are Async methods
+            // you may have noticed that we need to resort to clunky implementations for async operations
+            // (they're right above)
+            // so for now we will just take the folder path and access the files that way
+            auto path = winrt::to_string(foundFolder.Path());
+            path.append(FragmentsSubDirectory);
+            const auto jsonFiles = _AccumulateJsonFilesInDirectory(til::u8u16(path));
+
+            // Provide the package name as the source
+            _AddOrModifyProfiles(jsonFiles, ext.Package().DisplayName().c_str());
+        }
+    }
+}
+
+// Method Description:
+// - Helper function to apply json stubs in the local app data folder and the global program data folder
+// Arguments:
+// - The directory to find json files in
+// - The set of ignored namespaces
+void CascadiaSettings::_ApplyJsonStubsHelper(const std::wstring_view directory, const std::unordered_set<std::wstring>& ignoredNamespaces)
+{
+    const std::filesystem::path root{ wil::ExpandEnvironmentStringsW<std::wstring>(directory.data()) };
+
+    // The json files should be within subdirectories where the subdirectory name is the app name
+    for (auto& protoExtFolder : std::filesystem::directory_iterator(root))
+    {
+        auto folderPath = protoExtFolder.path().generic_string();
+
+        // We only want the parent folder name as the source (not the full path)
+        auto source = til::u8u16(folderPath.substr(folderPath.find_last_of("/") + 1));
+
+        // Only apply the stubs if the parent folder name is not in ignored namespaces
+        // (also make sure this is a directory for sanity)
+        if (std::filesystem::is_directory(protoExtFolder) && ignoredNamespaces.find(source) == ignoredNamespaces.end())
+        {
+            const auto jsonFiles = _AccumulateJsonFilesInDirectory(protoExtFolder.path().c_str());
+            _AddOrModifyProfiles(jsonFiles, winrt::hstring{ source });
+        }
+    }
+}
+
+// Method Description:
+// - Finds all the json files within the given directory
+// Arguments:
+// - directory: the directory to search
+// Return Value:
+// - A set containing all the found file data
+std::unordered_set<std::string> CascadiaSettings::_AccumulateJsonFilesInDirectory(const std::wstring_view directory)
+{
+    std::unordered_set<std::string> jsonFiles;
+
+    // Expand out environment strings like %LOCALAPPDATA% and %ProgramData%
+    const std::filesystem::path root{ wil::ExpandEnvironmentStringsW<std::wstring>(directory.data()) };
+
+    for (auto& protoExt : std::filesystem::directory_iterator(root))
+    {
+        if (protoExt.path().extension() == jsonExtension)
+        {
+            wil::unique_hfile hFile{ CreateFileW(protoExt.path().c_str(),

```

Comment:
What's checking here for failure to open the file? Just `if (hFile)`?

Do we want to check last error and throw it if we can't open it so it could be reported to the user somehow? (Or at least log it?)

Reply:
We throw the last error now!